### ISSUE TITLE
Enable librina and rinad tests as part of the continuous integration build process

### ISSUE THREAD

**Issue by miguelpdl** [IRATI/stack#1159]
Given that the [IRATI Stack now builds on the Travis CI](https://github.com/IRATI/stack/issues/1109) service the next step is to enable the librina and rinad tests to see how they support and assist the continuous development and deployment of the Stack.

**Comment by miguelpdl** [comment]
Created a new branch called [1159-check-functional](https://github.com/IRATI/stack/tree/1159-check-functional) and enabled the `make check-functional` in librina.

**Comment by miguelpdl** [comment]
Testing in the librina area on a Debian build machine and got a Segmentation fault:
`/bin/bash: line 5:   340 Segmentation fault      ${dir}$tst`

@edugrasa  would you mind taking a look at this testing issue for librina.

The larger set of messages around this segmentation fault included:

```
TESTING KMSG IPCM MEDIA REPORT
Serialized message size: 94
Test ok!
/bin/bash: line 5:   340 Segmentation fault      ${dir}$tst
FAIL: test-parsers
TESTING CONCURRENCY WRAPPER CLASSES
Concurrency: 0
Thread id: 3068299008
363(1511185447)#librina.concurrency (CRIT): Cannot join thread. Error is: 35
Caught expected exception. Cannot join thread
Created thread 0 with id 3068291904
Thread 0 started work
Created thread 1 with id 3059899200
Created thread 2 with id 3051506496
Created thread 3 with id Thread Thread 3 started work
3043113792
Thread 1 started work
2 started work
Created thread 4 with id 3034721088
Completed join with thread 0 having a status of 0
Completed join with thread 1 having a status of 0x1
Completed join with thread 2 having a status of 0x2
Completed join with thread 3 having a status of 0x3
Thread 4 started work
Completed join with thread 4 having a status of 0x4
Created thread 0 with id 3034721088
Created thread 1 with id 3043113792
Incremented counter, current value: Created thread 2 with id 3051506496
1Created thread 3
Incremented counter, current value: 2
 with id 3059899200
Created thread 4 with id 3068291904
Completed join with thread 0 having a status of 0
Completed join with thread 1 having a status of 0
Incremented counter, current value: 3
Incremented counter, current value: 4
Incremented counter, current value: 5
Completed join with thread 2 having a status of 0
Completed join with thread 3 having a status of 0
Completed join with thread 4 having a status of 0
```

**Comment by miguelpdl** [comment]
@edugrasa there does not appear to be any tests for rinad.

There's no test folder in the project, nor is there a Makefile option in the configure script to run. Is this the correct assessment ?

**Comment by edugrasa** [comment]
Hi Miguel,

Just fixed the librina and rinad tests, they should work if you merge from the latest arcfire branch. To run the rinad tests just run make check.

**Comment by miguelpdl** [comment]
Ran the tests for librina and rinad. 

rinad passed.

However the [librina check is failing](https://travis-ci.org/IRATI/stack/builds/309019308) with the following:

```
TESTING KMSG IPCP UPDATE CRYPTO STATE
Expected (172) and actual (148) message sizes are different
FAIL: test-parsers
```

I've ran this 3 different times on the Travis CI build server but had the same result, with this failure. 

I have tried on the Debian build machine and of course it passes there with no problem

```
TESTING KMSG IPCP UPDATE CRYPTO STATE
Serialized message size: 148
Test ok!
``` 

So for some reason on the Travis CI server (Ubuntu) has `Expected (172)` which is clearly not right.

The [test is this line here `test_irati_kmsg_ipcp_update_crypto_state()` in librina/test/test-parsers.cc](https://github.com/IRATI/stack/blob/76f7d0681aafc9eeaa38b823a3b5290657a15e85/librina/test/test-parsers.cc#L1564).

**Comment by vmaffione** [comment]
Updated travis build process: https://travis-ci.org/IRATI/stack

**Comment by edugrasa** [comment]
@miguelpdl that test was a successfull one, it uncovered a bug :) It is fixed now, make check-functional on librina passes all tests in Ubuntu 16.04 now

**Comment by miguelpdl** [comment]
@vmaffione ahh okay I didn't realise the extent of the travis build process changes until I reviewed all the commits made on Nov 30, 2017, but I've now got them on my branch.

@edugrasa I'm absolutely over the moon that this process has spotted a bug .... makes it worth while putting it in place. 

I've merged all the latest changes with the arcfire branch onto this branch and added to @vmaffione  to make sure that the tests are run in librina and rinad. That [build also passes on the Travis CI](https://travis-ci.org/IRATI/stack/builds/311053622) and also [passes on the GitLab Runner](https://gitlab.com/arcfire/irati/stack/-/jobs/42967602) so [created a PR on this work now](https://github.com/IRATI/stack/pull/1179).

**Comment by edugrasa** [comment]
Fixed by #1179
